include/xtensor/xview.hpp

```diff
@@ -50,33 +185,235 @@ namespace xt
+            static constexpr bool is_range_slice = is_xrange<slice>::value;
+            static constexpr bool is_int_slice = std::is_integral<slice>::value;
+            static constexpr bool is_all_slice = is_xall_slice<slice>::value;
+
+            static constexpr bool have_int_seen = int_seen || is_int_slice;
+
+            static constexpr bool is_valid = valid && (have_int_seen ? is_int_slice : (!range_seen && (is_all_slice || is_range_slice)));
+            static constexpr bool value = is_contiguous_view_impl<layout_type::column_major,
+                                                                  is_valid,
+                                                                  have_int_seen,
+                                                                  is_range_slice || range_seen,
+                                                                  xtl::mpl::pop_front_t<V>>::value;
+        };
+
+        template <bool valid, bool int_seen, bool range_seen>
+        struct is_contiguous_view_impl<layout_type::column_major, valid, int_seen, range_seen, xtl::mpl::vector<>>
+        {
+            static constexpr bool value = valid;
+        };
+
+
+        // TODO relax has_data_interface constraint here!
+        template <class E, class... S>
+        struct is_contiguous_view
+            : std::integral_constant<bool, 
+                has_data_interface<E>::value &&
+                !(E::static_layout == layout_type::column_major && static_dimension<typename E::shape_type>::value != sizeof...(S)) &&
+                is_contiguous_view_impl<E::static_layout, true, false, false, xtl::mpl::vector<S...>>::value
+              >
+        {
+        };
+
+        template <layout_type L, class T, std::ptrdiff_t offset>
+        struct unwrap_offset_container
+        {
+            using type = void;
+        };
+
+        template <class T, std::ptrdiff_t offset>
+        struct unwrap_offset_container<layout_type::row_major, T, offset>
+        {
+            using type = container_offset_view<T, offset>;
+        };
+
+        template <class T, std::ptrdiff_t start, std::ptrdiff_t end, std::ptrdiff_t offset>
+        struct unwrap_offset_container<layout_type::row_major, container_offset_view<T, start, end>, offset>
+        {
+            using type = container_offset_view<T, start + offset, end>;
+        };
+
+        template <class T, std::ptrdiff_t offset>
+        struct unwrap_offset_container<layout_type::column_major, T, offset>
+        {
+            using type = container_offset_view<T, 0, static_dimension<T>::value - offset>;
+        };
+
+        template <class T, std::ptrdiff_t start, std::ptrdiff_t end, std::ptrdiff_t offset>
+        struct unwrap_offset_container<layout_type::column_major, container_offset_view<T, start, end>, offset>
+        {
+            using type = container_offset_view<T, start, end - offset>;
+        };
+
+        template <class E, class... S>
```

Comment:
Add a comment that the slices are only `integers` and `all` with integers at the front.